app/routers/user.py

```diff
@@ -59,3 +59,38 @@ def get_all_user_events(session: Session, user_id: int) -> List[Event]:
         session.query(Event).join(UserEvent)
         .filter(UserEvent.user_id == user_id).all()
     )
+
+
+def disable_user(session: Session, user_id: int) -> bool:
+    """this functions changes user status to disabled.
+    returns:
+    True if function worked properly
+    False if it didn't."""
+
+    events_user_owns = list(session.query(Event).filter_by(owner_id=user_id))
+    # if user owns any event, he won't be able to disable himself.
+    if events_user_owns:
+        return False
+
+    # if user doesn't own any event,
+    # we will disable him and remove the user from all of its future events.
+    user_disabled = session.query(User).get(user_id)
+    user_disabled.disabled = True
+    future_events_for_user = session.query(UserEvent.id).join(
+        Event, Event.id == UserEvent.event_id
+        ).filter(
+        UserEvent.user_id == user_id, Event.start > datetime.now()).all()
+    # the query above finds events that user participates in the future.
+    for event_connection in future_events_for_user:
+        session.delete(event_connection)
```

Comment:
We shouldn't delete the user's events, he might want to get back in a minute

Reply:
removed this part of the function. Added a func that returns all available users